File: website/docs/guides/authentication/basic.py
# skip:start
from typegraph import typegraph, Policy, t, Graph
from typegraph.graph.params import Auth, Cors
from typegraph.runtimes.deno import DenoRuntime

# skip:end


@typegraph(
    auths=[
        # highlight-next-line
        Auth.basic(["admin"]),
    ],
    # skip:next-line
    cors=Cors(allow_origin=["https://metatype.dev", "http://localhost:3000"]),
)
def basic_authentication(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    ctx = t.struct({"username": t.string().optional().from_context("username")})

    g.expose(
        public,
        get_context=deno.identity(ctx),
    )

File: website/typegraphs/basic.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { Auth } from "@typegraph/sdk/params.js";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.js";

// skip:end

typegraph({
  name: "basic-authentication",
}, (g) => {
  const deno = new DenoRuntime();
  const pub = Policy.public();

  const ctx = t.struct({
    "username": t.string().optional().fromContext("username"),
  });

  // highlight-next-line
  g.auth(Auth.basic(["admin"]));

  g.expose({
    get_context: deno.identity(ctx).withPolicy(pub),
  });
});
